src/main/java/com/github/cameltooling/lsp/internal/completion/CamelOptionValuesCompletionsFuture.java

```diff
@@ -89,6 +90,33 @@ public List<CompletionItem> apply(CamelCatalog camelCatalog) {
 				}
 				return items;
 			}
+
+			//Check based on the value of the parameter
+			//if we get here, all the previous checks are done,
+			// so we can return it directly
+			return getCompletionItemsForParameterValue();
+		}
+		return Collections.emptyList();
+	}
+
+	private List<CompletionItem> getCompletionItemsForParameterValue() {
```

Comment:
I think the name is too generic, it will be more readable to mention something like getCompletionForKubernetesSecret